### ISSUE TITLE
Display Locking Up

### ISSUE THREAD

**Issue by JimPowersDFL** [fuel9/DisplayMonkey#135]
Hello, 

We are now running production for 22 displays. Most of them are Samsung smart TVs, using the built in browser. 15 of these displays must be on 24 hours a day. We are an animal shelter and these displays are showing personal protection equipment requirements for contagion control, so failure is not an option. We started seeing issues with these 15 displays randomly locking up and not updating (in fact, requiring a power cycle to reset). I have the same model at my desk and this morning, mine was locked up so it is duplicatable. 

Our server is 2012R2 so it isn't the random issue happening with 2016. The server is running in Hyper-V on my workstation (I have a 8-core system with 32GB of ram so speed isn't the issue and I'm using the high performance power plan). We'll eventually move it to our ESX environment but I'm not convinced that will make a difference. 

What are some things I can look at to help determine where the problem is?

**Comment by JimPowersDFL** [comment]
To add to this, this morning when I came in, I noticed the management side was running a bit sluggishly. When I looked at the server, there was an IIS worker process that was consuming 80% of the CPU. IIS hung when I tried to restart it so I had to manually kill that worker process. I did notice that the application pool was set to refresh every 29 hours (not sure where that came from). I changed that to 12 hours. Other ideas for this problem?

**Comment by Gumbees** [comment]
@JimPowersDFL we are experiencing the same mess. So much so we took this off of AWS. 

So for the performance of DisplayMonkey is horrible. The application has to render things directly from the database instead of storing them as files like everyone else does. This design in my opinion is very flawed. We are currently looking for other solutions.

**Comment by JimPowersDFL** [comment]
Well, I forgot to follow up on this particular issue. As far as the sluggish behavior of the management side, that only seems to happen when we haven't used it for some period of time (not sure how long). Once it wakes itself up, it performs as expected. I did upgrade to the latest version and haven't had any further issues with IIS processes locking up. We also moved the server off of my workstation to ESX which helped tremendously.

As far as the TVs go, I had a discussion with our vendor that installed them and they suggested that the problem is likely due to the fact that most of the displays we installed are consumer grade and not designed to be on all the time. I was able to convince the opening managers that they need to restart these TVs in the morning so I can have them auto power off. Since then, I haven't had any problems with them.

We're definitely learning as we go but I think we are in good shape now. Marketing is even happy with the system which was my biggest concern.
